server/server.go

```diff
@@ -21,14 +21,21 @@ import (
 	sqle "github.com/dolthub/go-mysql-server"
 )
 
+type ServerEventListener interface {
+	ClientConnected()
+	ClientDisconnected()
+	QueryStarted()
```

Comment:
Don't you need more info here to tell queries apart? Imagining you want to be able to show average execution time, top N slow queries, etc